source/agora/app.d

```diff
@@ -12,12 +20,53 @@
 *******************************************************************************/
 
 module agora.app;
+import agora.common.Config;
+import agora.common.Hash;
+import agora.node.Node;
+
+import vibe.core.args;
+import vibe.core.core;
+import vibe.core.log;
+import vibe.http.router;
+import vibe.http.server;
+import vibe.web.rest;
 
-void main()
+import std.digest.sha;
+import std.stdio;
+import core.stdc.time;
+import core.time;
+
+/// todo: work around this for 'dub test'
+version (unittest) void main () { } else:
```

Comment:
Shouldn't be necessary if you specify `mainSourceFile` in `dub.json`

Reply:
I tried doing this:

```

"configurations": [
    {
        "name": "agora",
        "targetName": "agora",
        "targetType": "executable",
        "mainSourceFile": "source/agora/app.d",
    },
    {
        "name": "unittest",
        "sourceFiles": [
            "source/scpp/build/DSizeChecks.o",
            "source/scpp/build/DLayoutChecks.o"
        ],
        "excludedSourceFiles": [ "source/agora/app.d" ]
    }
],
```

But then I get a linker error with `dub test`.